aiida/orm/nodes/process/process.py

```diff
@@ -479,7 +479,15 @@ def is_valid_cache(self):
 
         :returns: True if this process node is valid to be used for caching, False otherwise
         """
-        return super().is_valid_cache and self.is_finished
+        if not (super().is_valid_cache and self.is_finished):
+            return False
+        # Catch errors when getting the process_class, or when it does not have
+        # a 'is_valid_cache' method.
+        try:
+            is_valid_cache_func = self.process_class.is_valid_cache
```

Comment:
Can't this also throw other errors, specifically something like `EntryPointError` if the class cannot be loaded?

Reply:
Hmm yes, indeed it can. It's not quite clear to me what the correct response to that is, though. If the process class can not be loaded due to an entry point error, essentially we can no longer perform the checks that _might_ be implemented in that class. Should that fall back to being a valid cache, or invalid one?

Reply:
Should we perhaps err on the safe side and consider it to be invalid. If the user really want them to be cached they should simply install the relevant plugin. I am just wondering if in this case we should emit a warning otherwise these calculations may be silently skipped for caching and might be difficult to track down

Reply:
So we keep catching ``ValueError`` and return ``True`` (because that also happens in calcfunctions, I think), and for the rest log a warning and return ``False``?

Side note: This seems like a pretty weird edge case -- you're just launching a new process, but then its process class somehow isn't available. How does this even happen? Not saying we shouldn't handle it, though.

Reply:
Updated the code now.

Reply:
Seems I have misunderstood this before: The ``ValueError`` happens whenever the ``process_class`` can not be loaded.

This clashes with the existing calcfunctions tests not because calcfunctions are inherently problematic, but because those tests define new calcfunctions inside the class. Since these can not be imported, they are not allowed to cache. That could be fixed by moving all calcfunctions to the module level.

Since calcfunctions should anyway be importable for the daemon to find them, I think it is a safe choice to disable caching when they are not importable. It is definitely a change in behavior though, so if you want to be strict on backwards-compatibility, it should be enabled instead.